dev-client/src/components/common/search/search.ts

```diff
@@ -0,0 +1,34 @@
+import {useMemo, useState} from 'react';
+
+type TextSearchOptions<K extends string, T extends Record<K, string>> = {
+  data: T[];
+  keys: K[];
+};
+
+const normalizeText = (text: string) =>
+  text
+    .trim()
+    .toLowerCase()
+    .normalize('NFD')
+    .replace(/[\u0300-\u036f]/g, ''); // unicode range for combining diacritical marks
```

Comment:
i stole this from the web client! it took me a bit to understand how it works, but basically the default argument to `normalize` is `'NFC'`, which will give the shortest possible unicode form (i.e. if there is a specific code point for a letter with an accent, it will return that single codepoint, rather than the codepoint for the letter without the accent followed by the codepoint for the accent). `'NFD'` will instead give the longest possible form, so all letters with accents will get expanded into the letter's codepoint followed by the accent's codepoint. and then we remove all the codepoints for accents so our text search doesn't care about accents! no idea how well this works for non-latin alphabets :upside_down_face: